velox/row/UnsafeRowDeserializers.h

```diff
@@ -528,6 +532,23 @@ struct UnsafeRowPrimitiveBatchDeserializer {
   static StringView deserializeStringView(std::string_view data) {
     return StringView(data.data(), data.size());
   }
+
+  /// Origins from java side BigInteger(byte[] var).
+  /// The data is assumed to be in <i>big-endian</i> byte-order: the most
+  /// significant byte is the element at index 0.  The data is assumed to be
+  /// unchanged.
+  static int128_t deserializeLongDecimal(std::string_view data) {
+    const uint8_t* bytesValue = reinterpret_cast<const uint8_t*>(data.data());
+    int length = data.size();
+    uint8_t bytes[16]{};
+    for (int i = length - 1; i >= 0; --i) {
+      bytes[length - 1 - i] = bytesValue[i];
```

Comment:
Can we use byteswap here?

Reply:
See https://github.com/facebookincubator/velox/blob/main/velox/dwio/common/IntDecoder.h#L352

Reply:
Thanks! Changed to byteSwap.